src/intervals/arithmetic/trigonometric.jl

```diff
@@ -5,15 +5,31 @@
 # helper functions
 
 function _quadrant(x::AbstractFloat)
-    r = rem2pi(x, RoundNearest)
-    -2r > π && return 2 # [-π, -π/2)
-    r   < 0 && return 3 # [-π/2, 0)
-    2r  < π && return 0 # [0, π/2)
-    return 1 # [π/2, π]
+    PI_LO, PI_HI = bounds(bareinterval(typeof(x), π))
+
+    rlo = rem2pi(x, RoundDown) # [0, 2π]
```

Comment:
I don't think `rem2pi` can be assumed to give correct, or even faithful rounding. 

For `BigFloat` it is defined by
```julia
rem2pi(x::BigFloat, r::RoundingMode) = rem(x, 2*BigFloat(pi), r)
```
which I don't think would guarantee the correct rounding since $2\pi$ is not computed to any  higher precision and also not rounded in any specific direction. There is even a TODO-comment
```julia
# TODO: use a higher-precision BigFloat(pi) here?
```
So the one who implemented most likely didn't prove that it would be correct. I'm not sure how easy it would be to find a counter example though.

For `Float64` it has a quite complicated implementation and it probably gives very good results in general. But it was also easy to find a counterexample to the correctness. If we define
```julia
g = (rng, N) -> begin
    for _ in 1:N
        x = reinterpret(Float64, rand(rng, UInt))
        rem2pi(x, RoundDown) == Float64(rem2pi(big(x), RoundDown)) || error("x = $x")
    end
end
```
then it was enough to run
```julia
g(MersenneTwister(42), 1000)
```
to get the counterexample `x = 8.197839233774979e66`. For this `x` we have that the correctly rounded result should be `1.9664659792957133`, but `rem2pi(x)` gives us `1.9664659792957135` (so not even faithfully rounded).

I don't think this should be regarded as a bug in Julia though, most numerical methods don't give correct rounding.

Reply:
This issue was of course already present in the previous implementation. It is maybe reasonable to merge this PR, which fixes some issues, and leave this issue for another day.

Reply:
Of course it is probably extremely rare that any rounding errors in `rem2pi` actually affects the returned quadrant. It wouldn't surprise me if it actually gives correct results for all `Float64` values.

Reply:
Thx for peeking at the code for `rem2pi`. Yes perhaps this can be addressed in a separate PR. Then perhaps I should go back to using `rem2pi(..., RoundNearest)` since the version in this PR is flawed too, and less performant.

Reply:
To get a correct implementation I believe the easiest, but certainly not the fastest, is to more or less compute the remainder fully in interval arithmetic. Then you get a proper enclosure of it at least, if that enclosure is not enough to determine the quadrant you can fall back to some more pessimistic version. This would guarantee faithful, but not necessarily exactly correct determination of the quadrant. 

For `Float64` there is an implementation in CRlibm [here](https://github.com/taschini/pycrlibm/blob/master/crlibm/rem_pio2_accurate.c). This is likely very fast and would also guarantee a correct result. I'm not sure if it is part of the public interface though...

Reply:
Ah great thx for the reference.

I'll merge this PR (which now only focuses on thread-safety) and open an issue for `_quadrant` not being reliable.